source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
What about OpImageTexelPointer, OpCopyObject, OpSelect or OpPhi?
The "A pointer can only be an operand to the following instructions" list is incomplete too. This is missing all of the atomic functions for example.

I think both of these lists shouldn't really exist and any restrictions should either be explicitly listed for each opcode or should be implied through capabilities.

Quite unfortunate about bitcast though.

> Actually, the Vulkan spec does say that Logical must be selected for graphics modules.

Ah, thanks.